redhat-marketplace/redhat-marketplace-operator
# Add the ability for run metering on CRC

will set emptyDir on the prometheus deployment if defined on the meterbase cr

How to test:
- start a fresh CRC with 12 GiB of memory: `crc start -c 8 -m 12288`
- `make skaffold-dev` as normal
-   `k edit MeterBase rhm-marketplaceconfig-meterbase` 
and replace: 
```
  prometheus:
    resources: {}
    storage:
      size: 20Gi
```

with: 

```
promtheus:
  storage:
    emptyDir:
      medium: ""
      sizeLimit: "5Gi"
```

you should see all pods are running: 
```
prometheus-operator-67cbf74bb8-kb2hm           2/2     Running   0          40m
prometheus-rhm-marketplaceconfig-meterbase-0   5/5     Running   1          17m
prometheus-rhm-marketplaceconfig-meterbase-1   5/5     Running   1          17m
redhat-marketplace-operator-667b9f7cf4-j9dgw   1/1     Running   0          44m
rhm-metric-state-d4fb845d6-lzqcx               3/3     Running   0          40m
```
